### DISCUSSION TITLE
signIn function redirects to api/auth/error

### DISCUSSION THREAD

**Discussion by decidev22** [nextauthjs/next-auth#8012] [Help]
### Question 💬

I am writing a simple web app and successfully finished the part for the registration of a user.
The part where I am having an issue is the sign-in feature.
When I click on the sign-in button, it should just show me a toast of success or failure, but instead, it redirects me to "http://localhost:3000/api/auth/error"
This happens when...

- I enter the correct email and password to sign in
- I enter the wrong credential
- I click on the GitHub sign in option. (I have the correct id and secret)
- I click on Google sign in option

Is this expected behaviour?
Any help would be much appreciated.

My folder structure is...
main
-app
--(site)
---components
----AuthForm.tsx
----AuthSocialButton.tsx
-api
--auth
---[...nextauth]
----route.ts
--register
---route.ts

and here is a snippet of a code from my AuthForm.tsx.

```
const onSubmit: SubmitHandler<FieldValues> = (data) => {
    setIsLoading(true);
    if (variant === "REGISTER") {
      axios
        .post("/api/register", data)
        .catch(() => toast.error("Something went wrong"))
        .finally(() => setIsLoading(false));
    }

    if (variant === "LOGIN") {
      //from CredentialsProvider
      signIn("credentials", {
        ...data,
        redirect: false,
      })
        .then((callback) => {
          if (callback?.error) {
            toast.error("Invalid credentials provided");
          }
          if (callback?.ok && !callback?.error) {
            toast.success("Successfully logged in!");
          }
        })
        .finally(() => setIsLoading(false));
    }
  };
  const socialAction = (action: string) => {
    setIsLoading(true);
    signIn(action, { redirect: false }).then((callback) => {
      if (callback?.error) {
        toast.error("Invalid credential");
      }
      if (callback?.ok && !callback?.error) {
        toast.success("Successfully logged in!");
      }
    });
  };
```

### How to reproduce ☕️

Here is the branch where I am having an error.
https://github.com/decidev22/joggers-messenger/tree/feat/LoginFunc

### Contributing 🙌🏽

Yes, I am willing to help answer this question in a PR

**Comment by vedovato** [comment]
any update on this? happens to me if i hit the signIn button twice and quickly on mobile device

**Comment by muhaimincs** [comment]
what version are you using?

**Reply by vedovato** [reply]
next and next-auth latest

**Reply by decidev22** [reply]
I was also on the latest at the time of posting this. "next-auth": "^4.22.1"
